src/app/screens/PeerlyScreens/AppreciationScreen/appreciation.hooks.ts

```diff
@@ -1 +1,81 @@
-export function demohook() {}
+import {useMutation, useQuery} from 'react-query';
+import {
+  getCoreValuesList,
+  getCoworkerList,
+  postAppreciationRequest,
+} from '../../../services/PeerlyServices/appreciation';
+import {
+  PostAppreciationRequestBody,
+  GetCoworkersListRequest,
+  UserDetails,
+  CoreValue,
+} from '../../../services/PeerlyServices/appreciation/types';
+import toast from '../../../utils/toast';
+import {AxiosError} from 'axios';
+import {APIError} from './types';
+
+export function useGetCoworkerList(payload: GetCoworkersListRequest) {
+  const {data, isLoading, isError} = useQuery({
+    queryKey: ['coworker_list'],
+    queryFn: () => getCoworkerList(payload),
+    onError: (error: AxiosError<APIError>) => {
+      if (error.response?.data.message) {
+        toast(error.response.data.message, 'error');
+      } else {
+        toast('Something went wrong while fetching co-workers list', 'error');
+      }
+    },
+  });
+
+  let coworkerList = data?.data?.user_list?.map((item: UserDetails) => {
+    return {
+      label: `${item.first_name} ${item.last_name}`,
+      value: item.id,
+    };
+  });
+
+  return {data: coworkerList || [], isLoading, isError};
+}
+
+export function useGetCoreValuesList() {
+  const {data, isLoading, isError} = useQuery({
+    queryKey: ['profile_core_values'],
+    queryFn: getCoreValuesList,
+    onError: (error: AxiosError<APIError>) => {
```

Comment:
This would be an axios error in most cases but it might be a different error.

we can also handle error in the following way
```
onError: (error: unknown) => {
   if (error instance of AxiosError) {
       // use axios error
   } else {
       // generic error case
   }
}
```

Reply:
As per our discussion current handling is good. so no need of any chagnes.